components/form-control/script.js

```diff
@@ -0,0 +1,313 @@
+     */
+    serializeRangeSliderData(element, params) {
+        if (element.hasAttribute('two-handles')) {
+            console.warn('gameface-rangeslider component does not support form data when "two-handles" attribute is used!');
+            return;
+        }
+
+        //The real rangeslider value is hidden inside _value[0]
+        if (element.hasAttribute('name') && element._value[0] !== undefined) {
+            params.append(element.getAttribute('name'), element._value[0]);
+        }
+    }
+
+    /**
+     * Will serialize the data from gameface-checkbox component
+     * @param {HTMLElement} element - The gameface-checkbox element
+     * @param {URLSearchParams} params
+     */
+    serializeCheckboxData(element, params) {
+        if (!element.state.checked || !element.hasAttribute('name')) return;
+
+        params.append(element.getAttribute('name'), element.value);
+    }
+
+    /**
+     * Will serialize the data from gameface-switch component
+     * @param {HTMLElement} element - The gameface-switch element
+     * @param {URLSearchParams} params
+     */
+    serializeSwitchData(element, params) {
+        if (!element.checked || !element.hasAttribute('name')) return;
+        const value = element.getAttribute('value') || 'on';
+
+        params.append(element.getAttribute('name'), value);
+    }
+
+    /**
+     * Will serialize the data from gameface-dropdown component
+     * @param {HTMLElement} element - The gameface-dropdown element
+     * @param {URLSearchParams} params
+     */
+    serializeDropDownData(element, params) {
+        if (!element.hasAttribute('name')) return;
+
+        const dataName = element.getAttribute('name');
+        const selectedOptions = element.selectedOptions;
+
+        if (!selectedOptions.length) return;
+
+        //TODO: Make dropdown component to ignore selected options that are disabled if there is such a case
+        //By standard selected options that are disabled should not be added to the form data even they are selected in a multiple select.
+        for (let option of selectedOptions) {
+            params.append(dataName, option.value);
+        }
+    }
+
+    /**
+     * Will serialize the data from gameface-radiogroup component
+     * @param {HTMLElement} element - The gameface-radiogroup element
+     * @param {URLSearchParams} params
+     */
+    serializeRadionGroupData(element, params) {
+        const checkedButton = element.previouslyCheckedElement;
+        if (!checkedButton || !element.hasAttribute('name')) return;
+
+        params.append(element.getAttribute('name'), checkedButton.value);
+    }
+
+    /**
+     * Will serialize the data from form element
+     * @param {HTMLElement} element
+     * @param {URLSearchParams} params
+     */
+    serializeElementData(element, params) {
+        if (element.hasAttribute('disabled')) return;
+        const tagName = element.tagName.toLowerCase();
+
+        switch (tagName) {
```

Comment:
How we will add serialize methods to the input, button and textarea? For the rest gameface components I agree with you

Reply:
The serialization of the native elements is the most straight forward:
```
if (element.hasAttribute('name') && element.value !== undefined) {
      element.getAttribute('name'), element.value;
}
```

All other serialize<Element>Data methods have some additional logic that at the end comes to element.hasName, element.getValue, but in some other component-specific form. If we take advantage of the flexibility that custom elements provide, we could modify the value and name setters of each custom component to implement a standard API where you call element.value and it returns it. This way we'll be able to call it for all elements despite their type - custom or native.

For example:
We'll use the standard elements as a canon and try to make all of our custom elements behave the same way. So wee need to be able to get `name` and `value`.
Name's already standard, so we don't have to do anything. The value is a problem.

- RangeSlider - instead of checking element._value[0] - change the RangeSlider.value getter to access _valuie[0] internally. the two handles check can be moved to a validation method.
- Checkbox - instead of getting `element.state.checked` implement a value getter that will return it; again the validation can me moved to the checkbox
- Switch - add a value getter that will return element.getAttribute('value') || 'on'; if the default value 'on' is required only in a Form, the [component should check if it is a child of a form](https://github.com/chromium/chromium/blob/c4d3c31083a2e1481253ff2d24298a1dfe19c754/third_party/blink/renderer/core/html/custom/element_internals.cc#L346)
- Dropdown - make the value getter return the list of selectedOptions' values instead of doing in in the  serializeDropDownData
- RadioGroup - make a value getter that returns checkefButton.value

I understand that none of the above could be achieved without having the validation implemented first, so I think it is ok for this task. But I think that our approach is what makes this more complicated that it should be - instead of trying to workaround the limitations of the native elements, we should utilize the strengths of the custom elements to make them more "standard".